File: src/main/java/dev/vrba/simp/command/fun/animals/endpoints/DucksApiEndpoint.java
package dev.vrba.simp.command.fun.animals.endpoints;

import dev.vrba.simp.command.fun.animals.AnimalApiEndpoint;
import org.jetbrains.annotations.NotNull;
import reactor.core.publisher.Mono;

import java.io.InputStream;
import java.util.Set;

public class DucksApiEndpoint extends AnimalApiEndpoint {

    @Override
    @NotNull
    public Set<String> getNames() {
        return Set.of("duck", "ducks", "quack", "quacks");
    }

    @Override
    @NotNull
    public String getUrl() {
        return "https://random-d.uk/api/v1/random?type=png";
    }

    @Override
    @NotNull
    public String getTitle() {
        return "Quack!";
    }

    @Override
    @NotNull
    public Mono<String> extractImageFromResponse(@NotNull Mono<InputStream> response) {
        return this.extractImageFromJson(response, json -> json.get("url").asText());
    }
}

File: Services/ImageApi/Endpoints/DucksApiEndpoint.cs
using System.Net.Http.Json;

namespace SimpBot.Services.ImageApi.Endpoints;

public class DucksApiEndpoint : ImageApiEndpoint
{
    public override IEnumerable<string> Names { get; } = new[] {"duck", "ducks", "quack", "quacks"};

    public override string Url => "https://random-d.uk/api/v1/random?type=png";

    public override async Task<string?> ExtractImageUrlAsync(HttpResponseMessage response)
    {
        var data = await response.Content.ReadFromJsonAsync<IDictionary<string, string>>();
        var image = data?["url"];

        return image;
    }
}